tests/meta/demultiplex/test_demux_post_processing.py

```diff
@@ -774,3 +774,68 @@ def test_post_processing_of_flow_cell_demultiplexed_with_bclconvert(
     )
 
     delivery_path.exists()
+
+
+def test_post_processing_of_flow_cell_demultiplexed_with_bclconvert_flat_output_structure(
+    demultiplex_context: CGConfig,
+    flow_cell_directory_name_demultiplexed_with_bcl_convert_flat: str,
+    flow_cell_name_demultiplexed_with_bcl_convert: str,
+    demultiplexed_flow_cells_directory: Path,
+    bcl_convert_demultiplexed_flow_cell_sample_ids: List[str],
+):
+    # GIVEN a DemuxPostProcessing API
+    demux_post_processing_api = DemuxPostProcessingAPI(demultiplex_context)
+
+    # GIVEN a directory with a flow cell demultiplexed with BCL Convert
+    demux_post_processing_api.demux_api.out_dir = demultiplexed_flow_cells_directory
+
+    # WHEN post processing the demultiplexed flow cell
+    demux_post_processing_api.finish_flow_cell_temp(
+        flow_cell_directory_name_demultiplexed_with_bcl_convert_flat
+    )
+
+    # THEN a flow cell was created in statusdb
+    assert demux_post_processing_api.status_db.get_flow_cell_by_name(
+        flow_cell_name_demultiplexed_with_bcl_convert
+    )
+
+    # THEN sequencing metrics were created for the flow cell
+    assert (
+        demux_post_processing_api.status_db._get_query(table=SampleLaneSequencingMetrics)
+        .filter(
+            SampleLaneSequencingMetrics.flow_cell_name
+            == flow_cell_name_demultiplexed_with_bcl_convert
+        )
+        .all()
+    )
```

Comment:
As part of https://github.com/Clinical-Genomics/project-planning/issues/474 we are trying to not have sql alchemy code outside of dedicated filter functions. 

We should add filters if we need to retrieve something from a database like this.

Reply:
I dislike adding logic in our code base which is only used in tests

Reply:
I'll consider rewriting the test, whether this logic can be omitted or simplified

Reply:
The filter function already exists, it was implemented as part of https://github.com/Clinical-Genomics/cg/pull/2202 in a separate PR. #2202 is in hold due to the validation of the finish flow cell command.  You can add the filter then, it will not be dead-code.

Reply:
Added by Chris on other PR. I think it makes sense to add the filter functions in this specific case, where the usage will be implemented in the coming months.